src/Aspire.Hosting.Azure/Provisioning/Provisioners/AzureProvisioner.cs

```diff
@@ -94,6 +94,14 @@ async Task UpdateStateAsync(IAzureResource resource, Func<CustomResourceSnapshot
             {
                 await notificationService.PublishUpdateAsync(child, stateFactory).ConfigureAwait(false);
             }
+
+            // Some IAzureResource instances are a surrogate for for another resource in the app model
+            // to ensure that resource events are published for the resource that the user expects
+            // we lookup the resource in the app model here and publish the update to it as well.
+            if (appModel.Resources.Where(r => r.Name == resource.Name).SingleOrDefault() is { } nonAzureResource)
+            {
+                await notificationService.PublishUpdateAsync(nonAzureResource, stateFactory).ConfigureAwait(false);
```

Comment:
Need to keep track of the resource and its surrogate from the beginning so we don't need to do this search each time we fire the event.